src/Uno.UI/UI/Xaml/UIElement.RoutedEvents.cs

```diff
@@ -674,6 +674,15 @@ internal bool RaiseEvent(RoutedEvent routedEvent, RoutedEventArgs args, Bubbling
 				throw new InvalidOperationException($"Flag not defined for routed event {routedEvent.Name}.");
 			}
 
+			if (this is Popup popup)
+			{
+				// When an open flyout receives an event, the event propagation starts at
+				// the popup itself, and not it's (logical) child, so we need to
+				// handle this case explicitly. Note that the logical child isn't
+				// actually a child, so the event won't propagate back to the popup.
+				return popup.Child.RaiseEvent(routedEvent, args, ctx);
```

Comment:
For a KeyDown/Up. Currently, the element the routing starts with is FocusManager.GetFocusedElement(), which returns the popup if it's open. From my testing on windows, only the FlyoutPresenter fires my KeyDown callback when I subscribe to it.

Reply:
I've tested again today, and I'm not sure what the problem was, but the findings in the above comment are actually very incorrect. @MartinZikmund found that when the presenter is loaded, the popup switches focus to the the presenter and then the FocusManager will do some stuff so that in the end, the first focusable element inside the presenter is focused as intended. I've made some changes and things should be working now.